File: 2047A-A. Alyona and a Square Jigsaw Puzzle.py
def alyona_happy_days(t, test_cases):
    results = []
    
    for case in test_cases:
        n, pieces = case
        happy_days = 0
        total_pieces = 0
        layer_size = 1
        current_layer_pieces = 8
        
        for day_pieces in pieces:
            total_pieces += day_pieces
            
            while total_pieces >= layer_size:
                if total_pieces == layer_size:
                    happy_days += 1
                layer_size += current_layer_pieces
                current_layer_pieces += 8  # Increase the next layer size
        
        results.append(happy_days)
    
    return results

# Input Reading
t = int(input())
test_cases = []

for _ in range(t):
    n = int(input())
    pieces = list(map(int, input().split()))
    test_cases.append((n, pieces))

# Solve and Print Results
results = alyona_happy_days(t, test_cases)
for result in results:
    print(result)

File: 2047A-A. Alyona and a Square Jigsaw Puzzle.cpp
#include <iostream>
#include <vector>
using namespace std;

void alyonaHappyDays(int t, vector<pair<int, vector<int>>> &testCases) {
    for (int i = 0; i < t; ++i) {
        int n = testCases[i].first;
        vector<int> pieces = testCases[i].second;

        int happyDays = 0;
        int totalPieces = 0;
        int layerSize = 1; // Current layer size to complete
        int nextLayerIncrement = 8;

        for (int j = 0; j < n; ++j) {
            totalPieces += pieces[j];

            while (totalPieces >= layerSize) {
                if (totalPieces == layerSize) {
                    ++happyDays;
                }
                layerSize += nextLayerIncrement;
                nextLayerIncrement += 8; // Next layer has 8 more pieces than the current one
            }
        }

        cout << happyDays << endl;
    }
}

int main() {
    int t;
    cin >> t;
    vector<pair<int, vector<int>>> testCases;

    for (int i = 0; i < t; ++i) {
        int n;
        cin >> n;
        vector<int> pieces(n);
        for (int j = 0; j < n; ++j) {
            cin >> pieces[j];
        }
        testCases.push_back({n, pieces});
    }

    alyonaHappyDays(t, testCases);

    return 0;
}
